src/redis-check-aof.c

```diff
@@ -146,156 +266,412 @@ int readArgc(FILE *fp, long *target) {
     return readLong(fp,'*',target);
 }
 
-off_t process(FILE *fp) {
+int processRESP(FILE *fp, char *filename, int *multi) {
     long argc;
-    off_t pos = 0;
-    int i, multi = 0;
     char *str;
 
-    while(1) {
-        if (!multi) pos = ftello(fp);
-        if (!readAnnotations(fp)) break;
-        if (!readArgc(fp, &argc)) break;
-
-        for (i = 0; i < argc; i++) {
-            if (!readString(fp,&str)) break;
-            if (i == 0) {
-                if (strcasecmp(str, "multi") == 0) {
-                    if (multi++) {
-                        ERROR("Unexpected MULTI");
-                        break;
-                    }
-                } else if (strcasecmp(str, "exec") == 0) {
-                    if (--multi) {
-                        ERROR("Unexpected EXEC");
-                        break;
-                    }
+    if (!readArgc(fp, &argc)) return 0;
+
+    for (int i = 0; i < argc; i++) {
+        if (!readString(fp, &str)) return 0;
+        if (i == 0) {
+            if (strcasecmp(str, "multi") == 0) {
+                if ((*multi)++) {
+                    ERROR("Unexpected MULTI in AOF %s", filename);
+                    zfree(str);
+                    return 0;
+                }
+            } else if (strcasecmp(str, "exec") == 0) {
+                if (--(*multi)) {
+                    ERROR("Unexpected EXEC in AOF %s", filename);
+                    zfree(str);
+                    return 0;
                 }
             }
-            zfree(str);
-        }
-
-        /* Stop if the loop did not finish */
-        if (i < argc) {
-            if (str) zfree(str);
-            break;
         }
+        zfree(str);
     }
 
-    if (feof(fp) && multi && strlen(error) == 0) {
-        ERROR("Reached EOF before reading EXEC for MULTI");
-    }
-    if (strlen(error) > 0) {
-        printf("%s\n", error);
-    }
-    return pos;
+    return 1;
 }
 
-int redis_check_aof_main(int argc, char **argv) {
-    char *filename;
-    int fix = 0;
+int processAnnotations(FILE *fp, char *filename, int last_file) {
```

Comment:
please add a top comment explaining what it does.
what the return value means and in which cases it exists rather than return.
please do a similar thing to other functions

Reply:
OK